oauthproxy.go

```diff
@@ -277,6 +282,51 @@ func buildSignInMessage(opts *options.Options) string {
 	return msg
 }
 
+// buildRoutesAllowlist builds a []route from either the legacy SkipAuthRegex
+// option (paths only support) or newer SkipAuthRoutes option (method=path
+// support)
+func buildRoutesAllowlist(opts *options.Options) ([]*allowedRoute, error) {
```

Comment:
I moved configuration out of the options validation (to further the goal of that purely validating and doing no setup).

I know this isn't the best spot, but it gets it with the other very related configuration builders in this file that will eventually be refactored together in another PR to a better spot.